modules/carto/src/layers/spatial-index-tile-layer.js

```diff
@@ -0,0 +1,47 @@
+import {TileLayer} from '@deck.gl/geo-layers';
+
+function isFeatureIdDefined(value) {
+  return value !== undefined && value !== null && value !== '';
+}
+
+export default class SpatialIndexTileLayer extends TileLayer {
+  _updateAutoHighlight(info) {
+    const {hoveredFeatureId} = this.state;
+    const hoveredFeature = info.object;
+    let newHoveredFeatureId;
+
+    if (hoveredFeature) {
+      newHoveredFeatureId = hoveredFeature.id;
+    }
+    let {highlightColor} = this.props;
+    if (typeof highlightColor === 'function') {
```

Comment:
I feel we can move this piece of code inside of the next if `(hoveredFeatureId !== newHoveredFeatureId)` to save some performance